src/ucc/api/ucc.h

```diff
@@ -659,6 +660,48 @@ typedef struct ucc_oob_coll {
 
 typedef ucc_oob_coll_t ucc_context_oob_coll_t;
 typedef ucc_oob_coll_t ucc_team_oob_coll_t;
+
+/**
+ *
+ *  @ingroup UCC_CONTEXT_DT
+ */
+typedef enum  {
+    UCC_MEM_CONSTRAINT_SYMMETRIC   = UCC_BIT(0),
+    UCC_MEM_CONSTRAINT_PERSISTENT  = UCC_BIT(1),
+    UCC_MEM_CONSTRAINT_ALIGN32     = UCC_BIT(2),
+    UCC_MEM_CONSTRAINT_ALIGN64     = UCC_BIT(3),
+    UCC_MEM_CONSTRAINT_ALIGN128    = UCC_BIT(4),
+} ucc_mem_constraints_t;
+
+/**
+ *
+ *  @ingroup UCC_CONTEXT_DT
+ */
+typedef enum {
+    UCC_MEM_HINT_REMOTE_ATOMICS    = 0,
+    UCC_MEM_HINT_REMOTE_COUNTERS
+} ucc_mem_hints_t;
```

Comment:
is it correct to set both values simultaneously? in that case it should be UCC_BIT(0) and UCC_BIT(1)